### ISSUE TITLE
Store applications outside VectorCloud project

### ISSUE THREAD

**Issue by irongomme** [rmountjoy92/VectorCloud#3]
What do you think about storing applications in another repository(ies), to let contributors manage new vector applications or applications updates in an external space for it ?
In this way, application import could be loaded from a dedicated catalog, without storing them in this web interface repo.

**Comment by irongomme** [comment]
I have some ideas : 

1 - I've opened a ticket in anki forum : https://forums.anki.com/t/community-scripts-hub/25692 :
Then VectorCloud could load apps from an official community catalog

2 - Another way could be having another github project (Vector scripts hub) and use it as a catalog : 
Storing each github vector script project as a submodule. In this way, each project could be added with pull requests. VectorCloud could scan all vector apps in this new github project and sync regularly.

I start the debate, maybe someone else could have better solutions for that ;-)

**Comment by rmountjoy92** [comment]
yeah the only thing that makes me a little nervous about making it publicly editable is god forbid someone wanna be hacker replaces a line of code with something destructive. It could go unnoticed, someone could download it, run it and brick their computer.

**Comment by irongomme** [comment]
You mean that you could be responsible of the contributors packages ?

**Comment by rmountjoy92** [comment]
Nah I'm saying there would need to be some kind of authentication system where only the person that added the package can edit it and newly submitted apps need to be checked to make sure their not malicious.

**Comment by irongomme** [comment]
For exemple, I like the idea of Ansible Galaxy. Ansible is a server automation tool, and everyone can share small piece of logic to add into a personnal project, and this is done with dedicated website : https://galaxy.ansible.com
Everery share is done by linking a personnal github repo in. Then other users can star, see the code, download code, etc... 
I think this would be a great system for vector packages

**Comment by irongomme** [comment]
Hi, I'm learning flask, and i've began a project to accomplish this feature : https://github.com/irongomme/heyvector

For the beginning I've planned to : 
- let users publish their scripts 
- expose an url to let other tools like yours to import scripts

This not yet ready, but soon ! ;-)

**Comment by rmountjoy92** [comment]
Awesome! This is exciting! So if your application could send back a dictionary-like object for each application with the following fields:
    id = unique key e.g. 1
    script_name = application name
    author = application author
    website = author website
    description = application description
    icon = icon file in package
    download_link = where I could have VectorCloud look for the .zip
and I could use script_name = app_from_heyvector.script_name
It would be super easy to implement this.

**Comment by irongomme** [comment]
Yep, that could be something like that ! 
I'm thinking about some requirements to restrict package imports, like include a meta file in the each github project (like your ini file in each script exemple). Like that, I could easily distribute it in a json object.
